packages/vm/src/imports.rs

```diff
@@ -96,27 +97,26 @@ pub fn do_canonicalize_address<A: Api, S: Storage, Q: Querier>(
         Err(_) => return Ok(write_to_contract::<S, Q>(ctx, b"Input is not valid UTF-8")?),
     };
     let human: HumanAddr = source_string.into();
-    match api.canonical_address(&human) {
-        Ok(canon) => {
-            write_region(ctx, destination_ptr, canon.as_slice())?;
-            Ok(0)
-        }
-        // This check is unrelyable since we cannot tell by the error type if the error should be
-        // reported to the contract or indicates a broken backend.
-        // Err(FfiError::Other { error, .. }) => Ok(write_to_contract::<S, Q>(ctx, &error.as_bytes())?),
-        Err(error) => Err(error.into()),
-    }
+
+    let (canonical, gas_info) = api.canonical_address(&human)?;
```

Comment:
This is related to our other discussions about errors returned from this API.
I think the match was there because we were gonna later match on the `Err(error)` to see if it's something we want to show the contract.
Obviously if we don't show the contract anything here, then `?` does the same with less code, but i still think we should be checking this.

Reply:
Jap. Let's get back to this in the other ticket. We'll need to rewrite the match statement anyways since gas then needs to be processed in both success and error cases.

Reply:
hmm, that's a good point.

Reply:
in SecretNetwork we pass gas as an out-parameter (`&mut u64`) in a couple places. that might be a useful approach for that.